robolectric-resources/src/main/java/org/robolectric/res/PackageResourceIndex.java

```diff
@@ -0,0 +1,74 @@
+package org.robolectric.res;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+public class PackageResourceIndex implements ResourceIndex {
+  private static final Logger LOGGER = Logger.getLogger(ResourceExtractor.class.getName());
+
+  private final BiMap<Integer, ResName> resourceTable = HashBiMap.create();
+
+  private Integer maxUsedInt = null;
+  private Integer generatedIdStart = null;
+  private String packageName;
+  private int packageIdentifier;
+
+  public PackageResourceIndex(String packageName) {
+    this.packageName = packageName;
+  }
+
+  @Override
+  public synchronized Integer getResourceId(ResName resName) {
+    Integer id = resourceTable.inverse().get(resName);
+    if (id == null && ("android".equals(resName.packageName) || "".equals(resName.packageName))) {
+      if (maxUsedInt == null) {
+        maxUsedInt = resourceTable.isEmpty() ? 0 : Collections.max(resourceTable.keySet());
+        generatedIdStart = maxUsedInt;
+      }
+      id = ++maxUsedInt;
```

Comment:
Done. It didn't really streamline the code, but at least we're using a consistent resource id generation scheme now.